cmd/config_file.go

```diff
@@ -0,0 +1,107 @@
+/*
+ * Ceph Nano (C) 2018 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Below main package has canonical imports for 'go get' and 'go build'
+ * to work with all other clones of github.com/ceph/cn repository. For
+ * more information refer https://golang.org/doc/go1.4#canonicalimports
+ */
+
+package cmd
+
+import (
+	"path"
+
+	"github.com/alecthomas/units"
+	"github.com/spf13/viper"
+)
+
+func readConfigFile(customFile ...string) {
+	setDefaultConfig()
+	if len(customFile) > 0 {
+		var filename = path.Base(customFile[0])
+		var fileDir = path.Dir(customFile[0])
+		viper.SetConfigFile(filename)
+		viper.AddConfigPath(fileDir)
+		err := viper.ReadInConfig() // Find and read the config file
+		// If there is no configuration file, that's an error
+		if err != nil {
+			panic(err)
+		}
+	} else {
+		viper.SetConfigName("cn")         // name of config file (without extension)
+		viper.AddConfigPath("/etc/ceph/") // path to look for the config file in
```

Comment:
This won't work since the `ceph.conf` lives inside the container and is not accessible from the host.

Reply:
This is the configuration file for nano. So that file is on the host side. I was wondering where to put it. Could be /etc/cn or /etc/ceph-nano/ or any path you'd like.

Reply:
Ahhh, I got confused. Typically CLI programs such as cn relies on `$HOME/.cn/`

Reply:
Ok you're calling it multiple times... Then I believe `/etc/cn` is acceptable.